motr/client_init.c

```diff
@@ -1250,8 +1268,25 @@ static int initlift_idx_service(struct m0_sm *mach)
 		rc = service->is_svc_ops->iso_init((void *)ctx);
 		m0_sm_group_lock(&m0c->m0c_sm_group);
 
-		if (rc != 0)
-			initlift_fail(rc, m0c);
+		if (rc != 0) {
+			/*
+			 * Added retry logic to handle out of
+			 * order startup of data and server
+			 * PODs. Ref: Jira ID Cortx-33899
+			 */
+			if (retry_count < MAX_CLIENT_INIT_RETRIES
+			    && (rc == -EIO || rc == -EPROTO)) {
+				retry_count += 1;
+				M0_LOG(M0_ERROR, "client init \
+				       failed with %d. Retrying.", rc);
+				return M0_RC(initlift_get_cur_floor(m0c));
+			} else {
+				retry_count = 0;
+				initlift_fail(rc, m0c);
+			}
+		} else {
+			retry_count = 0;
+		}
```

Comment:
I think this else block is not not needed.

Reply:
@madhavemuri This will be useful  if any phase after IDX fails, and if any application tries to retry m0_client_init().